src/binaryen-c.cpp

```diff
@@ -69,12 +69,14 @@ BinaryenLiteral toBinaryenLiteral(Literal x) {
       memcpy(&ret.v128, x.getv128Ptr(), 16);
       break;
     case Type::funcref:
-      ret.func = x.getFunc().c_str();
-      break;
-    case Type::nullref:
+      if (!x.isNull()) {
+        ret.func = x.getFunc().c_str();
+      }
```

Comment:
The payload's 16 bytes are initialized with zeros anyway in this case, so the `Name` `struct`s string will be `nullptr`. Can make this explicit if you prefer.

Reply:
Where is the payload initialized to zero? I'm pretty sure the declaration of `ret` leaves them uninitialized since `BinaryenLiteral` is a POD type because all its members are [trivially constructable](https://en.cppreference.com/w/cpp/language/default_constructor#Trivial_default_constructor).

Either way, probably better to make this explicit to be safe.

Reply:
Oh, you are right, this is the C-API struct. Confused this with `Literal` where the `: v128()` in the ctors initializes the payload to all zeroes in the other case. Going to fix.